db/config.cc

```diff
@@ -745,6 +745,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , user_defined_function_contiguous_allocation_limit_bytes(this, "user_defined_function_contiguous_allocation_limit_bytes", value_status::Used, 1024*1024, "How much memory each UDF invocation can allocate in one chunk")
     , schema_registry_grace_period(this, "schema_registry_grace_period", value_status::Used, 1,
         "Time period in seconds after which unused schema versions will be evicted from the local schema registry cache. Default is 1 second.")
+    , max_concurrent_cql_requests(this, "max_concurrent_cql_requests", value_status::Used, std::numeric_limits<uint64_t>::max(),
+        "Maximum number of concurrent CQL requests a single shard can handle before it starts shedding extra load. Default 0 means that no requests will be shed.")
```

Comment:
0 is outdated (I presume)
uint64_t is excessive.

Perhaps the name should emphasize the per-shardness. It's easy to overlook.

Reply:
It is outdated. I'll fix the description, the name, and switch to u32 tomorrow.

Reply:
@psarna @avikivity it has to have the same type as `_requests_serving` if we want to keep the `limits` trick. 
If we change `max_concurrent_cql_requests` to u32 - we need to change the ` _requests_serving` type too (and I guess we don't want to do that) or reconsider the default value the relevant logic. 

I think keeping `max_concurrent_cql_requests` as u64 despite that probably being excessive is an acceptable compromise.

Reply:
Well, it doesn't really *have to* be the same type as long as we agree that it's not realistic of  `_requests_serving` to ever grow beyond 4294967295. But on the other hand I agree that it makes more sense -- e.g. if a bug in `_requests_serving` causes it to grow indefinitely, `max_concurrent_cql_requests` would start acting really weird as well. I'm fine with rolling it back to `uint64_t`, but since @avikivity  requested the change I'll wait for his comment as well.

Reply:
I don't see a problem with u32. We can change _requests_serving to that type too if it makes anyone more comfortable, but it isn't really necessary.

Reply:
If we compare two integer types of different size the compiler will have to extend the smaller one anyway during the comparison. 

Also, since the parameter is not a naked uintX but a named_value<uintX> and the value itself is located after two string_view values and currently string_view has two following non-static fields:

```
const char* ptr_;
  size_type length_;
```

And unless I'm missing something...
No matter what integer type we are going to use it's going to be 64-bit aligned (since we do not use packing for named_value).

Therefore if we use uint64 or uint32 we are going to use/waste the same amount of memory.
Meaning using uint32 is not going to save any.

And then back to what @psarna wrote and what I meant all alone - using the same type for both ` _requests_serving` and `max_concurrent_cql_requests` while not a must is safer given the logic we intend to use.

Reply:
Oh, I noticed that I did missed an uint8_t value after the T in the named_value. 
So, using uint32 here would actually do save some space here. 

However, I still think that using uint64_t would be safer have.
And since we decided not to (as it seems) then we can no longer use the `limit::max()` trick here.